src/app/stripe.service.ts

```diff
@@ -55,35 +55,30 @@ export class StripeService {
     // checks that we are handling Stripe objects as intended.
     // See https://github.com/stripe/stripe-js
     this.stripe = await loadStripe(environment.psps.stripe.publishableKey);
-    if (this.stripe) {
-      this.elements = this.stripe.elements({fonts: [
-        {
-          family: 'Euclid Triangle',
-          src: `url('${environment.donateGlobalUriPrefix}/d/EuclidTriangle-Regular.1d45abfd25720872.woff2') format('woff2')`,
-          weight: '400',
-        },
-      ]});
-    }
   }
 
-  async createPaymentMethod(
-    cardElement: StripeCardElement,
-    donorName?: string,
-  ): Promise<{paymentMethod?: PaymentMethod; error?: StripeError}> {
+  stripeElements(donation: Donation, campaign: Campaign)
+  {
     if (!this.stripe) {
-      console.log('Stripe not ready');
-      return {};
+      throw new Error('Stripe not ready');
     }
 
-    // See https://stripe.com/docs/js/payment_methods
-    const result = await this.stripe.createPaymentMethod({
-      type: 'card',
-      card: cardElement,
-      billing_details: {
-        name: donorName,
-      },
+    const amountInMinorUnit = (donation.tipAmount + donation.donationAmount) * 100;
+
+    return this.stripe.elements({
+      fonts: [
+        {
+          family: 'Euclid Triangle',
+          src: `url('${environment.donateGlobalUriPrefix}/d/EuclidTriangle-Regular.1d45abfd25720872.woff2') format('woff2')`,
+          weight: '400',
+        },
+      ],
+      mode: 'payment',
+      currency: donation.currencyCode.toLowerCase(),
+      amount: amountInMinorUnit,
+      setup_future_usage: 'on_session',
+      on_behalf_of: campaign.charity.stripeAccountId,
```

Comment:
`on_behalf_of` has to be set here to match how it is already set when matchbot creates the payment intent, otherwise we get an error from stripe: `The provided on_behalf_of (acct_xxx) does not match the expected on_behalf_of (null). Try confirming with a Payment Intent that is configured to use the same parameters as Stripe Elements` when pressing the pay button.

Along with the future usage, it makes this message appear within the payment element:

![image](https://github.com/thebiggive/donate-frontend/assets/159481/26af5299-e7f5-4a54-b85d-cbde30b126f2)

Reply:
(discovered later that we can turn off display of the terms within the stripe payment element, e.g. if we want to rely on our own copy instead)